### ISSUE TITLE
Faces (fs) is undefined - Use in A-Frame

### ISSUE THREAD

**Issue by LukeTOBrien** [samalexander/three-csg-ts#17]
Hello,

I am using [Aframe-Typescript-Class-Components](https://www.npmjs.com/package/aframe-typescript-class-components) to write a subtraction component in A-Frame and I am [getting the ThreeJS object using their API](https://aframe.io/docs/1.2.0/introduction/developing-with-threejs.html#setting-an-object3d-on-an-entity).  
I am using the code in your `Example Usage` but at runtime I am getting the error `fs is undefined`.  
In your `index.ts` I can see that `fs` are the faces of the geometry:
```typescript
static fromGeometry(geom: BufferGeometry | Geometry): CSG {
    if (geom instanceof BufferGeometry) {
      geom = new Geometry().fromBufferGeometry(geom);
    }
    const fs = geom.faces; // <- Here. Faces is undefined
    ...
```

I have found [this answer on SO](https://stackoverflow.com/questions/42141438/access-to-faces-in-buffergeometry)

> BufferGeometries in three.js are fundamentally different from the regular Geometries. They are not oriented towards ease of manipulation but rather towards how meshes need to be delivered to the WebGL API.

It seems that not all meshes are created equally and the meshes created using A-Frame do not have faces property.  
Would you be able to experiment and do some testing with A-Frame?  
If you like I could create a repo.

Thanks.

The full Typescript code for my component is:
```typescript
import { Component, Entity, ObjectMap, System } from "aframe";
import { BaseComponent, toComponent } from "aframe-typescript-class-components";
import { CSG } from "three-csg-ts";

export interface SubtractData {
    type: string
}

export class SubtractComponent extends BaseComponent<SubtractData> {
    static schema = {
        type: { type: "string" as const, default: "" }
    };

    update() {
        let sibling = this.el.previousElementSibling as Entity<ObjectMap<Component<any, System<any>>>>;
        console.log(sibling);
        console.log(sibling.object3DMap);

        // Make 2 box meshes..
        const meshA = sibling.getObject3D("mesh") as THREE.Mesh;
        const meshB = this.el.getObject3D("mesh") as THREE.Mesh;

        // Make sure the .matrix of each mesh is current
        meshA.updateMatrix();
        meshB.updateMatrix();

        console.log(meshA);
        console.log(meshB);

        // Create a bsp tree from each of the meshes
        const bspA = CSG.fromMesh(meshA); // <- Error, mesh.geometry does not have faces
        const bspB = CSG.fromMesh(meshB);

        // Subtract one bsp from the other via .subtract... other supported modes are .union and .intersect
        const bspResult = bspA.subtract(bspB);

        // Get the resulting mesh from the result bsp
        const meshResult = CSG.toMesh(bspResult, meshA.matrix);

        // Set the results material to the material of the first cube.
        meshResult.material = meshA.material;
        
        sibling.setObject3D("mesh", meshResult);

    }
}
AFRAME.registerComponent("subtract", toComponent(SubtractComponent));
```
And html test page is:
```html
<head>
    <script src="https://cdn.jsdelivr.net/npm/aframe/dist/aframe-master.min.js"></script>
    <script type="text/javascript" src="./SUBTRACT_COMPONENTjs">
    </script>
  </head>
  <body>
    <a-scene>
      <a-entity id="rig" position="0 1.6 0">
        <a-entity id="camera" camera  look-controls wasd-controls="acceleration:100; fly: true"></a-entity>
      </a-entity>

      <a-box  rotation="0 45 0" color="#4CC3D9"></a-box>
      <a-sphere subtract radius="0.65" color="#EF2D5E"></a-sphere>
      
      <a-sky color="#ECECEC"></a-sky>
    </a-scene>
  </body>
```

**Comment by yandeu** [comment]
A-Frame v1.2.0 has updated to Three.js r125 and has therefore removed support for Geometry.

See [A-Frame changelog](https://github.com/aframevr/aframe/blob/master/CHANGELOG.md#120-feb-5-2021).

Similar to #16

**Comment by samalexander** [comment]
@LukeTOBrien could you confirm whether this is resolved in v3.0.0?
